Classes/BEMCheckBox.m

```diff
@@ -32,6 +33,18 @@ @interface BEMCheckBox ()
  */
 @property (strong, nonatomic) BEMPathManager *pathManager;
 
+/** The group this box is associated with.
+ */
+@property (weak, nonatomic, nullable) BEMCheckBoxGroup *group;
+
+@end
+
+/** Defines private methods that we can call to update our group's status.
+ */
+@interface BEMCheckBoxGroup ()
+
+- (void)_checkBoxSelectionChanged:(BEMCheckBox *)checkBox;
```

Comment:
Gotcha, thanks for the explanation. This is where Swift's internal/private/public cone handy.
I'm going to sleep over this one, and try to think if there would be a more elegant solution.